scipio/src/dma_file.rs

```diff
@@ -251,6 +257,31 @@ I will close it and turn a leak bug into a performance bug. Please investigate",
 }
 
 impl DmaFile {
+    /// Returns true if the files represent the same file on the underlying device.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use scipio::{LocalExecutor, DmaFile};
+    ///
+    /// let ex = LocalExecutor::make_default();
+    /// ex.run(async {
+    ///     let mut wfile = DmaFile::create("myfile.txt").await.unwrap();
+    ///     let mut rfile = DmaFile::open("myfile.txt").await.unwrap();
+    ///     // Different objects (OS file descriptors), so they will be different...
+    ///     assert!(wfile != rfile);
+    ///     // However they represent the same object.
+    ///     assert!(wfile.is_same(&rfile));
+    ///     wfile.close().await;
+    ///     rfile.close().await;
+    /// });
+    /// ```
```

Comment:
Good documentation but it would be befenicial to explain how doing this differs from comparing fds. Especially irt to symbolic links. Your existing documentation sorta does it already, but would gain from being more explicit.

Reply:
I am open to removing PartialEq.

I can also add a mention to symlinks (this would return false for a symlink).

Reply:
Both changes are now done.